packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -221,6 +285,8 @@ export function renderWithHooks(
           getComponentName(Component),
         );
       }
+      verifyHookCallOrder();
+      currentHookMatches = [];
```

Comment:
Both of these statements should be DEV-only?

Reply:
Seems like maybe we could avoid re-allocating `currentHookMatches` and just do `.length = 0` here.